### PR TITLE
Handle all hash codes equally

### LINKED ISSUE #4: Executor would skip tests if their hashCode turns out to be Integer.MIN_VALUE
See `line 22` https://github.com/alsutton/sharded-test-executor/blob/1b203db8bc7f93bce22f0e7c78353b5d1ae1b08f/lib/core/src/main/java/com/alsutton/shardedtestexecutor/TestExecutionPredicate.java#L21-L24

`Math.abs(int)` does not guarantee the result will be non-negative 🤷  (try it with `Integer.MIN_VALUE`)

### PR DESCRIPTION
**Pull request by ephemient**
Resolves #4

### CODE REVIEW

File: lib/core/src/main/java/com/alsutton/shardedtestexecutor/TestExecutionPredicate.java [hunk 1]
```diff
@@ -19,8 +19,8 @@ public TestExecutionPredicate() {
 
     @Override
     public boolean test(String testClassName) {
-        int classHash = Math.abs(classHasher.hash(testClassName));
-        int targetShard = classHash % environment.getTotalShardCount();
+        int classHash = classHasher.hash(testClassName);
+        int targetShard = Integer.remainderUnsigned(classHash, environment.getTotalShardCount());
```
Comment (vlsi) [inline lib/core/src/main/java/com/alsutton/shardedtestexecutor/TestExecutionPredicate.java:23]: An alternative approach is `java.lang.Math#floorMod(int, int)` (thanks to @amaembo in https://twitter.com/tagir_valeev/status/1408344589216399360)

Comment (ephemient) [reply lib/core/src/main/java/com/alsutton/shardedtestexecutor/TestExecutionPredicate.java:23]: Assuming positive shared count, Integer.remainderUnsigned or Math.floorMod have the same overall distribution across all hash codes, but the former may be handled as an intrinsic in some environments (https://twitter.com/jakewharton/status/1311678590715604993, https://bugs.openjdk.java.net/browse/JDK-8238669) while I've never heard of that being proposed for the latter.

Comment (vlsi) [reply lib/core/src/main/java/com/alsutton/shardedtestexecutor/TestExecutionPredicate.java:23]: Yeah, `remainderUnsigned` is probably easier to read: it is clear that the output is unsigned :)

**Comment by alsutton** [comment]
Thanks for the PR, this makes things cleaner.

**Pull request merged**